src/System.IO.UnmanagedMemoryStream/src/System/IO/UnmanagedMemoryStream.cs

```diff
@@ -429,34 +421,24 @@ public override long Position
             [System.Security.SecurityCritical]  // auto-generated_required
             get
             {
-                if (_buffer != null)
-                {
-                    throw new NotSupportedException(Strings.NotSupported_UmsSafeBuffer);
-                }
+                if (_buffer != null) throw new NotSupportedException(Strings.NotSupported_UmsSafeBuffer);
+                if (!_isOpen) throw __Error.GetStreamIsClosed();
 
                 // Use a temp to avoid a race
                 long pos = Interlocked.Read(ref _position);
-                if (pos > _capacity)
-                    throw new IndexOutOfRangeException(Strings.IndexOutOfRange_UMSPosition);
+                if (pos > _capacity) throw new IndexOutOfRangeException(Strings.IndexOutOfRange_UMSPosition);
                 byte* ptr = _mem + pos;
-                if (!_isOpen) throw __Error.GetStreamIsClosed();
+
                 return ptr;
             }
             [System.Security.SecurityCritical]  // auto-generated_required
             set
             {
-                if (_buffer != null)
-                    throw new NotSupportedException(Strings.NotSupported_UmsSafeBuffer);
+                if (_buffer != null) throw new NotSupportedException(Strings.NotSupported_UmsSafeBuffer);
                 if (!_isOpen) throw __Error.GetStreamIsClosed();
+                if (value < _mem) throw new IOException(Strings.IO_SeekBeforeBegin);
 
-                // Note: subtracting pointers returns an Int64.  Working around
-                // to avoid hitting compiler warning CS0652 on this line. 
-                if (new IntPtr(value - _mem).ToInt64() > UnmanagedMemStreamMaxLength)
-                    throw new ArgumentOutOfRangeException("offset", Strings.ArgumentOutOfRange_UnmanagedMemStreamLength);
-                if (value < _mem)
-                    throw new IOException(Strings.IO_SeekBeforeBegin);
-
-                Interlocked.Exchange(ref _position, value - _mem);
+                Seek(value - _mem, SeekOrigin.Begin);
```

Comment:
Thanks, @KrzysztofCwalina.  If after you speak with folks today the consensus is that the exceptional differences are ok, and if the changes to use Seek don't measurably and negatively affect perf, LGTM.